core/domain/stats_domain.py

```diff
@@ -41,6 +41,158 @@
 MIGRATED_STATE_ANSWER_TIME_SPENT_IN_SEC = 0.0
 
 
+class ExplorationStats(object):
+    """Domain object representing analytics data for an exploration."""
+
+    def __init__(self, exp_id, exp_version, num_actual_starts, num_completions,
+                 state_stats_mapping):
+        """Constructs an ExplorationStats domain object.
+
+        Args:
+            exp_id: str. ID of the exploration.
+            exp_version: int. Version of the exploration.
+            num_actual_starts: int. Number of learners who actually attempted
+                the exploration.
+            num_completions: int. Number of learners who completed the
+                exploration.
+            state_stats_mapping: dict. A dictionary mapping the state names of
+                an exploration to the corresponding StateStats domain object.
+        """
+        self.exp_id = exp_id
+        self.exp_version = exp_version
+        self.num_actual_starts = num_actual_starts
+        self.num_completions = num_completions
+        self.state_stats_mapping = state_stats_mapping
+
+    def to_dict(self):
+        """Returns a dict representation of the domain object."""
+        exploration_stats_dict = {
+            'exp_id': self.exp_id,
+            'exp_version': self.exp_version,
+            'num_actual_starts': self.num_actual_starts,
+            'num_completions': self.num_completions,
+            'state_stats_mapping': self.state_stats_mapping
+        }
+        return exploration_stats_dict
+
+    def validate(self):
+        """Validates the ExplorationStats domain object."""
+
+        if not isinstance(self.exp_id, basestring):
+            raise utils.ValidationError(
+                'Expected exp_id to be a string, received %s' % (self.exp_id))
+
+        if not isinstance(self.exp_version, int):
+            raise utils.ValidationError(
+                'Expected exp_version to be an int, received %s' % (
+                    self.exp_version))
+
+        if not isinstance(self.num_actual_starts, int):
+            raise utils.ValidationError(
+                'Expected num_actual_starts to be an int, received %s' % (
+                    self.num_actual_starts))
+
+        if not isinstance(self.num_completions, int):
+            raise utils.ValidationError(
+                'Expected num_completions to be an int, received %s' % (
+                    self.num_completions))
+
+        if not isinstance(self.state_stats_mapping, dict):
+            raise utils.ValidationError(
+                'Expected state_stats_mapping to be a dict, received %s' % (
+                    self.state_stats_mapping))
+
+
+class StateStats(object):
+    """Domain object representing analytics data for an exploration's state."""
+
+    def __init__(self, total_answers_count, useful_feedback_count,
```

Comment:
again, break on start of parentheses if you have to.

Reply:
Done.